### PR TITLE
Tidy up the BUILD rules in the sdk package

### PR DESCRIPTION
**Pull request by csilvestrini**
by moving all the kotlincopts into the standard bzl file

### CODE REVIEW

File: third_party/java/arcs/build_defs/internal/kotlin.bzl [hunk 1]
```diff
@@ -41,28 +41,54 @@ IS_BAZEL = not (hasattr(native, "genmpm"))
...
+    "PackageName",
+    "TopLevelName",
+]
+
+def _merge_lists(*lists):
+    result = {}
+    for l in lists:
+        for elem in l:
+            result[elem] = 1
+    return result.keys()
```
Comment (jasonwyatt) [inline third_party/java/arcs/build_defs/internal/kotlin.bzl]: No change needed, just want to gripe about how it's so silly that Starlark doesn't have "Set"s

File: third_party/java/arcs/build_defs/internal/kotlin.bzl [hunk 2]
```diff
@@ -41,28 +41,54 @@ IS_BAZEL = not (hasattr(native, "genmpm"))
...
+
 def arcs_kt_jvm_library(**kwargs):
-    """Declares arbitrary kotlin library targets for the JVM.
+    """Wrapper around kt_jvm_library for Arcs.
+
+    Provides a bunch of default arguments.
```
Comment (jasonwyatt) [inline third_party/java/arcs/build_defs/internal/kotlin.bzl]: This line doesn't really add anything helpful. It'd be nice to know what arguments it provides defaults for.

File: third_party/java/arcs/build_defs/internal/kotlin.bzl [hunk 3]
```diff
@@ -41,28 +41,54 @@ IS_BAZEL = not (hasattr(native, "genmpm"))
...
-        ] + disable_lint_checks
         kwargs["constraints"] = constraints
+        kwargs["disable_lint_checks"] = _merge_lists(disable_lint_checks, DISABLED_LINT_CHECKS)
 
     kt_jvm_library(**kwargs)
 
+def arcs_kt_native_library(**kwargs):
+    """Wrapper around kt_native_library for Arcs.
+
+    Provides a bunch of default arguments.
```
Comment (jasonwyatt) [inline third_party/java/arcs/build_defs/internal/kotlin.bzl]: Same, either describe the defaults it provides, or consider just removing the line.

File: third_party/java/arcs/build_defs/internal/kotlin.bzl (hunk 1 above)
Comment (csilvestrini) [reply third_party/java/arcs/build_defs/internal/kotlin.bzl]: Tell me about it... :roll_eyes:

File: third_party/java/arcs/build_defs/internal/kotlin.bzl (hunk 2 above)
Comment (csilvestrini) [reply third_party/java/arcs/build_defs/internal/kotlin.bzl]: I'm gonna delete it. It's annoying to keep up to date.

File: third_party/java/arcs/build_defs/internal/kotlin.bzl (hunk 3 above)
Comment (csilvestrini) [reply third_party/java/arcs/build_defs/internal/kotlin.bzl]: Same

**Pull request merged**